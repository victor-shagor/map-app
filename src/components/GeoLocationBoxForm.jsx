import React from "react";

import Button from "./Button";
import Input from "./Inputs";

const GeoLocationBoxForm = ({
  inputData,
  handleChange,
  onSubmit,
  isLoading,
}) => {
  return (
    <form className="form-div" onSubmit={onSubmit} data-testid="form">
      <Input
        id={"minlong"}
        name={"minlong"}
        label="Min Longitude"
        value={inputData.minlong}
        onChange={handleChange}
        min={"-180"}
        max={"180"}
        type="number"
      />
      <Input
        name={"minlat"}
        id={"minlat"}
        label="Min Latitude"
        value={inputData.minlat}
        onChange={handleChange}
        min={"-90"}
        max={"90"}
        type="number"
      />
      <Input
        name={"maxlong"}
        id={"maxlong"}
        label="Max Longitude"
        value={inputData.maxlong}
        onChange={handleChange}
        min={"-180"}
        type="number"
        max={"180"}
      />
      <Input
        name={"maxlat"}
        id={"maxlat"}
        label="Max Latitude"
        value={inputData.maxlat}
        onChange={handleChange}
        min={"-90"}
        type="number"
        max={"90"}
      />
      <Button text={"submit"} isLoading={isLoading} />
    </form>
  );
};

export default GeoLocationBoxForm;
