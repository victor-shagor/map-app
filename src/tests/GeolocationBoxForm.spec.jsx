import { fireEvent, render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";

import GeoLocationBoxForm from "components/GeoLocationBoxForm";

describe("GeoLocationBoxForm Component", () => {
  const onSubmit = jest.fn();
  const handleChange = jest.fn();
  const inputData = {
    minlong: 10,
    minlat: 20,
    maxlong: 10,
    maxlat: 20,
  };
  it("form values should be accurate", () => {
    render(
      <GeoLocationBoxForm
        inputData={inputData}
        onSubmit={onSubmit}
        handleChange={handleChange}
      />
    );
    expect(screen.getByTestId("form")).toHaveFormValues(inputData);
  });

  it("onChange shoould call handlechange function", async () => {
    render(
      <GeoLocationBoxForm
        inputData={inputData}
        onSubmit={onSubmit}
        handleChange={handleChange}
      />
    );
    const minlong = screen.getByLabelText("Min Longitude");
    fireEvent.change(minlong, { target: { value: 20 } });
    expect(handleChange).toHaveBeenCalled();
  });

  it("submit should call onsubmit function", async () => {
    render(
      <GeoLocationBoxForm
        inputData={inputData}
        onSubmit={onSubmit}
        handleChange={handleChange}
      />
    );
    const form = screen.getByTestId("form");
    fireEvent.submit(form);
    expect(onSubmit).toHaveBeenCalled();
  });

  it("renders correctly", () => {
    const tree = renderer
      .create(
        <GeoLocationBoxForm
          inputData={inputData}
          onSubmit={onSubmit}
          handleChange={handleChange}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
