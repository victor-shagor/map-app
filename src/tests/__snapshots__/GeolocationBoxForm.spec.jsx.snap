// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GeoLocationBoxForm Component renders correctly 1`] = `
<form
  className="form-div"
  data-testid="form"
  onSubmit={[MockFunction]}
>
  <div
    className="input-div"
  >
    <label
      htmlFor="minlong"
    >
      Min Longitude
    </label>
    <input
      className="input"
      id="minlong"
      max="180"
      min="-180"
      name="minlong"
      onChange={[MockFunction]}
      required={true}
      step="any"
      type="number"
      value={10}
    />
  </div>
  <div
    className="input-div"
  >
    <label
      htmlFor="minlat"
    >
      Min Latitude
    </label>
    <input
      className="input"
      id="minlat"
      max="90"
      min="-90"
      name="minlat"
      onChange={[MockFunction]}
      required={true}
      step="any"
      type="number"
      value={20}
    />
  </div>
  <div
    className="input-div"
  >
    <label
      htmlFor="maxlong"
    >
      Max Longitude
    </label>
    <input
      className="input"
      id="maxlong"
      max="180"
      min="-180"
      name="maxlong"
      onChange={[MockFunction]}
      required={true}
      step="any"
      type="number"
      value={10}
    />
  </div>
  <div
    className="input-div"
  >
    <label
      htmlFor="maxlat"
    >
      Max Latitude
    </label>
    <input
      className="input"
      id="maxlat"
      max="90"
      min="-90"
      name="maxlat"
      onChange={[MockFunction]}
      required={true}
      step="any"
      type="number"
      value={20}
    />
  </div>
  <button
    className="btn"
    disabled={false}
    type="submit"
  >
    submit
  </button>
</form>
`;
